File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.go
func numsSameConsecDiff(n int, k int) []int {
    nums := []int{1, 2, 3, 4, 5, 6, 7, 8, 9}
    
    for i := 1; i < n; i++ {
        nextNums := make([]int, 0, len(nums))
        
        for _, num := range nums {
            if plusK := (num % 10) + k; plusK < 10 {
                nextNums = append(nextNums, 10 * num + plusK)
            }
            if lessK := (num % 10) - k; lessK >= 0 && k != 0 {
                nextNums = append(nextNums, 10 * num + lessK)
            }
        }
        
        nums = nextNums
    }
    
    return nums
}

File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.py
class Solution:
    def numsSameConsecDiff(self, n: int, k: int) -> List[int]:
        nums = deque([1, 2, 3, 4, 5, 6, 7, 8, 9])

        for _ in range(n - 1):
            for each in range(len(nums)):
                num = nums.popleft()

                plus_k = (num % 10) + k
                less_k = (num % 10) - k

                if plus_k < 10:
                    nums.append(10 * num + plus_k)
                if less_k >= 0 and k != 0:
                    nums.append(10 * num + less_k)

        return nums
    